director/authentication.go

```diff
@@ -0,0 +1,40 @@
+package director
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/param"
+	log "github.com/sirupsen/logrus"
+)
+
+type DiscoveryResponse struct {
+	JwksUri string `json:"jwks_uri"`
+}
+
+// Returns links related to director's authentication, including the link
+// to get the public key from the director
+func discoveryHandler(ctx *gin.Context) {
+	director := param.Federation_DirectorUrl.GetString()
+	rs := DiscoveryResponse{
+		JwksUri: director + "/.well-known/public_signing_key.jwks",
```

Comment:
Three comments about this line:

I tested without setting the director URL, and we get here without a warning. When that happens, the jwksuri is just the path. We should add logic to make sure that doesn't happen. I think Brian B added a function somewhere in the last few weeks to generate a server's hostname -- in the event that no director URL is set, we should probably make sure to at least use that.

Second comment is easy -- can you look into whether these types of things should have a newline at the end? When I try curling the director for the key, I get a file that doesn't end in a newline, which goes against the POSIX standard (although I'm not confident that's something we need to follow here).

Third comment is also easy -- unless there's another reason I don't know about, we should probably stick to using `-` separators here instead of suddenly switching to `_`.

Reply:
For the first note, I throw 500 error instead of falling back to hostname just to be consistent with behaviors from other places.

For the third note, I was following the current osg-htc.org issuer's format where they have https://osg-htc.org/.well-known/pelican-configuration for discovery and https://osg-htc.org/osdf/public_signing_key.jwks for jwks. I can change to use `-` if we don't need to take consideration of this.